### ISSUE TITLE
Error propagation symbol (`?`) can only be used on `Result`s

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#10348]
## Environment

* **IntelliJ Rust plugin version:** 0.4.193.5328-231-nightly
* **Rust toolchain version:** 1.70.0-nightly (9df3a39fb 2023-04-11) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2023.1 Ultimate Edition (IU-231.8109.175)
* **Operating system:** macOS 12.6
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
False negative error detection for the `?` operator

## Steps to reproduce
There is a code
```rust
async fn test_inline() -> Result<(), Error> {
    let y = bar().await?;
}

async fn bar() -> Option<i32> {
    let a = 1;
    Some(a)
}
```
## Actual Results
there is no highlighting for ` bar().await?`

<img width="982" alt="Screenshot 2023-04-13 at 15 45 34" src="https://user-images.githubusercontent.com/1213122/231778354-5b4c6d9e-abab-4d3c-971d-a14098327c25.png">
